bot/exts/moderation/infraction/infractions.py

```diff
@@ -480,6 +474,47 @@ async def apply_ban(
         async def action() -> None:
             await ctx.guild.ban(user, reason=reason, delete_message_days=purge_days)
 
+        if isinstance(user, Member) and any(role.id in constants.STAFF_PARTNERS_COMMUNITY_ROLES for role in user.roles):
+            confirmation_view = _utils.StaffBanConfirmationView(
+                allowed_users=(ctx.author.id,),
+                allowed_roles=constants.MODERATION_ROLES,
+                timeout=10,
+            )
+
+            await confirmation_view.send(
+                ctx.channel,
+                message_content="That user is a Helper. Are you sure you want to ban them?",
```

Comment:
Probably want to use a general warning verbiage here as `STAFF_PARTNERS_COMMUNITY_ROLES` is more than just Helpers. I like that we warn for all of them.

```suggestion
                message_content=f"{user} has an elevated role. Are you sure you want to ban them?",
```